tests/test_simulate.py

```diff
@@ -107,33 +127,52 @@ def test_rf():
 @pytest.mark.xfail(not ADVANCED_DEPS["tensorflow"],
                    raises=ImportError,
                    reason="requires tensorflow")
-def test_nn_2_layer():
+def test_nn_2_layer(tmpdir):
+
+    # copy state file to tmp dir for changes
+    tmp_json_state_fp = Path(tmpdir, "tmp_state.json")
+    copyfile(json_state_file, tmp_json_state_fp)
+
     check_model(model="nn-2-layer",
-                state_file=json_state_file,
+                state_file=tmp_json_state_fp,
                 n_instances=1,
                 n_queries=2)
 
 
 @pytest.mark.xfail(not ADVANCED_DEPS["tensorflow"],
                    raises=ImportError,
                    reason="requires tensorflow")
-def test_lstm_base():
+def test_lstm_base(tmpdir):
+
+    # copy state file to tmp dir for changes
+    tmp_h5_state_fp = Path(tmpdir, "tmp_state.h5")
```

Comment:
Is there a reason some have h5 state files and some json? Just for variety?

Reply:
Yes, it isn't consistent. I think this is for variety. Might be good to test the complete array (in another PR). However, we will replace both state files with a new state model. So, it might not be very useful to prioritize this.